### PR TITLE
Implement examples hub

### PR DESCRIPTION
**Pull request by FejZa**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

Implement a simple `ExamplesHubSystem` that once registered with the root profile, will spawn a little UI with example selection. Examples can be configured via profiles. This is the beginning of efforts to create an XRTK examples hub application showcasing XRTK capabilities.

Examples can be configured to be only visible to selected platforms.

Depends on https://github.com/XRTK/XRTK-Core/pull/875

examples-hands-profile
examples-locomotion-profile
examples-hub-system-profile
examples-hub-system
examples-hub-ui

### CODE REVIEW

File: XRTK.Examples/Assets/XRTK.Examples/Editor/ExamplesHubSystemProfileInspector.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            }
+
+            rect.height = EditorGUIUtility.singleLineHeight;
+            rect.y += 3;
+            var exampleDataProperty = examples.GetArrayElementAtIndex(index);
+            var selectedExampleData = EditorGUI.ObjectField(rect, exampleDataProperty.objectReferenceValue, typeof(ExampleProfile), false) as ExampleProfile;
+
+            if (selectedExampleData != null)
+            {
+                selectedExampleData.ParentProfile = ThisProfile;
```
Comment (StephenHodgson) [inline XRTK.Examples/Assets/XRTK.Examples/Editor/ExamplesHubSystemProfileInspector.cs:71]: Is this the only place we use this?

File: XRTK.Examples/Assets/XRTK.Examples/Editor/ExamplesHubSystemProfileInspector.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        {
+            if (isFocused)
+            {
+                currentlySelectedExampleElementIndex = index;
+            }
+
+            rect.height = EditorGUIUtility.singleLineHeight;
+            rect.y += 3;
+            var exampleDataProperty = examples.GetArrayElementAtIndex(index);
+            var selectedExampleData = EditorGUI.ObjectField(rect, exampleDataProperty.objectReferenceValue, typeof(ExampleProfile), false) as ExampleProfile;
```
Comment (StephenHodgson) [inline XRTK.Examples/Assets/XRTK.Examples/Editor/ExamplesHubSystemProfileInspector.cs]: How come you couldn't use the default profile drawer?

**Review by StephenHodgson — CHANGES_REQUESTED** [review]
be sure to update the xrtk package reference once it get's published after the core PR is merged

File: XRTK.Examples/Assets/XRTK.Examples/Editor/ExamplesHubSystemProfileInspector.cs (hunk 2 above)
Comment (FejZa) [reply XRTK.Examples/Assets/XRTK.Examples/Editor/ExamplesHubSystemProfileInspector.cs]: image
This is what I get with the default profile drawer. Unfortunately not working.

**Comment by FejZa** [comment]
Update: I added a default ExamplesHubSystemProfile and made it part of the DemoMixedRealityRootProfile. That means loading hte base scene and pressing play, will spawn the Examples Hub UI by default.

Note: It is built in a way that it supports multi-scene editing in Unity. As soon as there is more than only the base scene loaded in Unity, the examples hub will not do anything and respect the developers scene setup. This is useful when you are working on a specific example and don't want to have to launch starting from the base scene every time.

**Comment by FejZa** [comment]
@StephenHodgson there was no 0.2.19 released for Core it seems. We have to merge to the main branch I guess?

**Comment by StephenHodgson** [comment]
@FejZa it's the 0.3.0-preview

File: XRTK.Examples/Assets/XRTK.Examples/Editor/ExamplesHubSystemProfileInspector.cs (hunk 2 above)
Comment (StephenHodgson) [reply XRTK.Examples/Assets/XRTK.Examples/Editor/ExamplesHubSystemProfileInspector.cs]: I'll take a look and see if I can get this sorted out

**Comment by StephenHodgson** [comment]
@FejZa I merged in the locomotion PR then updated this one, but I'd like to you confirm that this is updated correctly

**Comment by FejZa** [comment]
I pulled this out of the locomotion system PR. Since that one is merged this is obsolete. But I'll double-check.

**Pull request closed without merging**